### DISCUSSION TITLE
How does mutt generate the reply message body?

### DISCUSSION THREAD

**Discussion by zorghere** [neomutt/neomutt#4514] [Development]
Apologies if this has been covered, but I haven't been able to find an answer/solution to what I'm trying to achieve in the documentation or elsewhere in theg github issues.

I want to run a separate command on the original message body when viewing in the pager, and different command on the same message body when replying, to format the quoted body in the editor. This is because I'm converting some attachment types to text to preview in the pager, but I don't want these automatically included in the quoted message body when replying.

The only promising thing I could find about this is setting `edit=<command>` and `compose=<command>` in the mailcap entry for plain/text (in this case), but it seems to have no effect -- editing or composing a message always uses the editor as set in the neomutt configuration.

What am I missing?

**Comment by gahr** [comment]
The `pager` setting controls the command to be use to page (view) the message. It defaults to `builtin`, which means use the built-in pager in NeoMutt. Not sure if that helps / if I got the question right.

**Comment by zorghere** [comment]
That part I understand. I probably wasn't clear enough in my question, but what I'm asking is in two parts:

1. Does NeoMutt take the body of the message as displayed in the built-in pager (using mailcap rules) and then process that to generate new quote levels before sending it to the editor when replying? Or does it regenerate the body of the message using the same mailcap rules before sending to the editor?

2. Is there any way/setting to intervene to modify the message body as displayed in the pager when replying, so that the quoted message body which is sent to the editor is processed in some way (stripping certain words, for instance?)

The only way that occured to me feels very hacky and might not work depending on the answers to the above, but I thought that a `send-hook` that points to a different mailcap file with text/plain entry that processes the body could potentially work, but only I suppose if the reply body is generated again using the mailcap view command, rather than using what is already being displayed when reading the message in the built-in pager.

Would this work? Is there another way?

**Comment by gahr** [comment]
Oh, I got the question now. Perhaps we could look into using the `compose` field of mailcap when looking up a mailcap entry to use for reply. Currently, we only use it for the `new-mime` function to add new attachments when composing an email. That would be a breaking change, though.. 

Probably, setting mailcap on send-hook is how I'd work around it.

**Comment by nickwynja** [comment]
> Perhaps we could look into using the `compose` field of mailcap when looking up a mailcap entry to use for reply.

I'd find this useful! I find myself often wanting to reply to HTML emails for work and would like to retain the original HTML in the reply body. In this case, I'd probably change my reply format to be more like Outlook's where there's a horizontal rule and then the header information for attribution, the original message HTML (without `>` quote prefix / `<blockquote>`).

I've tried a `send-hook` workaround but am not getting very far.
